### PR TITLE
[fix] Fix bugs in debug_tools scripts

### PR DESCRIPTION
**Pull request by balazske**
The debug_tools did not work with current version of files got from
clang and tu_collector. At least part of the problems is fixed.
The problems are related to file format changes and name of clang executable.

### CODE REVIEW

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- scripts/debug_tools/prepare_compiler_info.py  1
- scripts/debug_tools/prepare_analyzer_cmd.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=6069485&bid=19875604)

File: scripts/debug_tools/prepare_all_cmd_for_ctu.py [hunk 1]
```diff
@@ -30,8 +30,8 @@ def execute(cmd):
             errors="ignore")
         out, err = proc.communicate()
 
-        print("stdout:\n\n" + out.decode("utf-8"))
-        print("stderr:\n\n" + err.decode("utf-8"))
+        print("stdout:\n\n" + out)
+        print("stderr:\n\n" + err)
```
Comment (martong) [inline scripts/debug_tools/prepare_all_cmd_for_ctu.py:34]: Is this a Python3 related change?

File: scripts/debug_tools/prepare_analyzer_cmd.py [hunk 2]
```diff
@@ -26,9 +26,18 @@ def __init__(self, opts):
 
     def __call__(self, path):
 
-        if re.search('clang$', path):
+        if re.search('clang(-(\d)+)?$', path):
```
Comment (martong) [inline scripts/debug_tools/prepare_analyzer_cmd.py:29]: Could you please add a comment with an example for the matches: `clang-7` 
Would it work with minor versions too? Could we have `clang-7.0.1` ?

File: scripts/debug_tools/prepare_compiler_info.py [hunk 3]
```diff
@@ -20,17 +20,19 @@ def prepare(compiler_info_file, sources_root):
...
-                lib.change_paths(line,
-                                 lib.IncludePathModifier(sources_root_abs)))
-        new_json_data[compiler] = {
-            'includes': changed_lines,
-            'target': json_data[compiler]['target'],
-            'default_standard': json_data[compiler]['default_standard']
-        }
+        new_json_data[compiler] = dict()
+        for language in json_data[compiler]:
+            lines = json_data[compiler][language]['compiler_includes']
```
Comment (martong) [inline scripts/debug_tools/prepare_compiler_info.py:25]: Ah, I remember when @gyorb added the `language` stuff... (that is the time when it broke, but we forgot then)

**Review by martong — COMMENTED** [review]
Do you think it would be write a test case that utilizes the tool? So we'd not find ourselves with a broken tool again. Could be in another patch.

File: scripts/debug_tools/prepare_all_cmd_for_ctu.py (hunk 1 above)
Comment (balazske) [reply scripts/debug_tools/prepare_all_cmd_for_ctu.py:34]: It looks like yes (but I am not sure).

File: scripts/debug_tools/prepare_compiler_info.py (hunk 3 above)
Comment (gyorb) [reply scripts/debug_tools/prepare_compiler_info.py:25]: Yes, probably the update for this script was left out at that time.

File: scripts/debug_tools/prepare_analyzer_cmd.py (hunk 2 above)
Comment (balazske) [reply scripts/debug_tools/prepare_analyzer_cmd.py:29]: I checked only in my build directory, and there is "clang-11" only. I think this is generated by default to differentiate between main releases.

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- scripts/debug_tools/prepare_compiler_info.py  1
- scripts/debug_tools/prepare_analyzer_cmd.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=6069485&bid=19875604)

**Review by martong — APPROVED** [review]
